### PR TITLE
Created SplashScreen and moved all logic to core

### PR DESCRIPTION
**Pull request by MKiperszmid**

### CODE REVIEW

File: app/src/main/java/com/mk/tasky/core/data/TaskyRepositoryImpl.kt [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
 import com.squareup.moshi.Moshi
 import retrofit2.HttpException
 import java.util.concurrent.CancellationException
 
-class AuthenticationRepositoryImpl(
-    private val api: AuthenticationApi
-) : AuthenticationRepository {
+class TaskyRepositoryImpl(
+    private val api: TaskyApi
+) : TaskyRepository {
```
Comment (philipplackner) [inline app/src/main/java/com/mk/tasky/core/data/TaskyRepositoryImpl.kt]: Why did you rename this? `AuthenticationRepository` is much more expressive than `TaskyRepository` (which screams to me "this is the repo where we put in all the stuff that's shared" 😆 )

File: app/src/main/java/com/mk/tasky/core/domain/preferences/Preferences.kt [hunk 2]
```diff
@@ -3,9 +3,9 @@ package com.mk.tasky.core.domain.preferences
 import com.mk.tasky.core.domain.model.LoggedUser
 
 interface Preferences {
-    fun saveToken(token: String)
-    fun saveFullName(name: String)
-    fun saveUserId(userId: String)
+    fun saveToken(token: String?)
+    fun saveFullName(name: String?)
+    fun saveUserId(userId: String?)
```
Comment (philipplackner) [inline app/src/main/java/com/mk/tasky/core/domain/preferences/Preferences.kt]: What's the point of saving null values here? In case that relates to clearing them, I'd prefer having a function `deleteToken()` etc.
`saveToken(null)` doesn't really read as remove the token, at least not at the first sight

**Review by philipplackner — APPROVED** [review]

File: app/src/main/java/com/mk/tasky/core/data/TaskyRepositoryImpl.kt (hunk 1 above)
Comment (MKiperszmid) [reply app/src/main/java/com/mk/tasky/core/data/TaskyRepositoryImpl.kt]: Yeah you are right.
Fixed!

File: app/src/main/java/com/mk/tasky/core/domain/preferences/Preferences.kt (hunk 2 above)
Comment (MKiperszmid) [reply app/src/main/java/com/mk/tasky/core/domain/preferences/Preferences.kt]: Fixed!

**Review by philipplackner — APPROVED** [review]
